DependencyInjection/SonataUserExtension.php

```diff
@@ -88,12 +88,12 @@ public function load(array $configs, ContainerBuilder $container)
         $profileFormDefinition = $container->getDefinition('sonata.user.profile.form');
         $registrationFormDefinition = $container->getDefinition('sonata.user.registration.form');
         if (method_exists($profileFormDefinition, 'setFactory')) {
-            $profileFormDefinition->setFactory(array('form.factory', 'createNamed'));
-            $registrationFormDefinition->setFactory(array('form.factory', 'createNamed'));
+            $profileFormDefinition->setFactory(array(new Reference('form.factory'), 'createNamed'));
+            $registrationFormDefinition->setFactory(array(new Reference('form.factory'), 'createNamed'));
         } else {
-            $profileFormDefinition->setFactoryClass('form.factory');
+            $profileFormDefinition->setFactoryClass(new Reference('form.factory'));
```

Comment:
Because it's called "factory class". I didn't check the docs, but I think it makes sense, right?